lua/lspkind/init.lua

```diff
@@ -63,8 +63,8 @@ local kinds = {
   },
   -- Kind items list
   list = {
-    cmp = require('vim.lsp.protocol').CompletionItemKind,
-    sym = require('vim.lsp.protocol').SymbolKind,
+    cmp = vim.lsp.protocol.CompletionItemKind,
+    sym = vim.lsp.protocol.SymbolKind,
```

Comment:
I don't think it's has something with _deep copying_. You should use `vim.deepcopy` for that. You can check if `vim.lsp.protocol.SymbolKind` hasn't been changed after `init` called via 
```:lua print(vim.inspect(require('vim.lsp.protocol').SymbolKind))```

Reply:
I don't understand what the point is, it is necessary that this table is changed, it is not necessary to make a copy, just a shorter reference

Reply:
You are changing vim internal table `vim.lsp.protocol.SymbolKind` which you shouldn't. You can change `cmp` table but currently it's **pointing/referencing** on const-like table.

Reply:
It is also necessary to change the `vim.lsp.protocol.SymbolKind` table symbols as it is also part of the lsp protocol. I see no point in changing one symbol table and leaving the other unchanged.
But it is also necessary to change the xxx table symbols as it is also part of the lsp protocol. I see no point in changing one symbol table and leaving the other unchanged.
If you read the commit message, you will see that the SymbolKind setup has been added

TIP: you can use `:lua vim.pretty_print(...)` to get same result as `:lua print(vim.inspect(...)`

Reply:
Wait, why do you need to change those table?

Reply:
> It is also necessary to change the vim.lsp.protocol.SymbolKind table symbols as it is also part of the lsp protocol.

What?! You are not supposed to change **anything** in **nvim** lsp protocol implementation. And there is no need for that to achieve plugin goals.

Reply:
`vim.lsp.protocol.SymbolKind` has similar usage to `vim.lsp.protocol.CompletionItemKind`, what would be the problem in changing it?

Reply:
Emm... I've already explained the issue with changing something that doesn't belong to the scope of this plugin, especially **core** things that are nvim internals. So the answer will be the same - do not change things that is not supposed to be changed. In any other strict programming language those tables would be constants and you will not be able to modify them by definition.

Reply:
> Emm... I've already explained the issue with changing something that doesn't belong to the scope of this plugin, especially **core** things that are nvim internals. So the answer will be the same - do not change things that is not supposed to be changed. In any other strict programming language those tables would be constants and you will not be able to modify them by definition.

Plugin scope is an easy way to define lsp kind symbols, isn't it?
Both tables I said are about lsp kind symbols
And also as you can see, the plugin already changed `CompletionItemKind` table, I just changed the way they are changed
Not changing these values ​​leaves the modified symbols restricted to the plugin's API, and changing any plugin or script that requires these tables will get the value that was set by setup, not the default value that the core provides.
I also reinforce that changing them will not break anything in the core, because it specifically changes the displayed symbols

Reply:
I agree that current implementation also changes `CompletionItemKind` - which is **wrong**, same as with changing **SymbolKind**. Just do `help vim.lsp.protocol` and read the first sentence:

```Module vim.lsp.protocol defines constants dictated by the LSP specification```

They are **constants**. Period. If you modify it you brake reverse lookup functionality that advertised few sentences later.

Reply:
Changing the values ​​does not break the reverse lookup, as the plugin only changes the symbols, for that reason I put the kinds.order table, so that it is certified that only the corresponding value changes, and the indexing values ​​remain unchanged
I understand that constants should not be changed, but if we want the plugin to modify the symbols and be compatible with other plugins possibly using this table
However, if I know of a better way to implement this I would like it to show

Reply:
> > Changing the values ​​does not break the reverse lookup
> 
> It does. On your branch `vim.lsp.protocol.SymbolKind[1] == "File"` is `false`, but it should be `true` according to official docs.

This is the expected behavior, because change the symbol
The same will happen if you try to vim.lsp.protocol.CompletionItemKind[1] == "Text", it will return false because the plugin changes the symbol too
If we don't want to change the internal tables, it will only be possible to use the custom symbols directly from the plugin's API, which are exposed in lspkind.symbols, and so the plugin doesn't change either CompletionItemKind or SymbolKind

> If you don't care about this breakage at least make it optional. For what ever reasons anyone wants it - OK, he can enable it. But for those who uses normal methods that doesn't require to change nvim internals - like `option 2` - make it work without that breakage.

I agree to make an option